### PR TITLE
Add RBAC check to issue_batch api

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* We currently perform this check on the bulk-issue page rendering, but we also need to check the API

**Release Note**

```release-note
RBAC check for BulkIssue on the API
```

### CODE REVIEW

File: pkg/controller/issueapi/issue_batch.go [hunk 1]
```diff
@@ -71,6 +72,13 @@ func (c *Controller) HandleBatchIssue() http.Handler {
 			return
 		}
 
+		if !membership.Can(rbac.CodeBulkIssue) {
```
Comment (sethvargo) [inline pkg/controller/issueapi/issue_batch.go:75]: I think we can only do this check if it was called via the UI. There is no "membership" on the API.

I'm also pretty sure I did this - isn't there a specific handler for the UI server to server this?

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: pkg/controller/issueapi/issue_batch.go (hunk 1 above)
Comment (whaught) [reply pkg/controller/issueapi/issue_batch.go:75]: No the UI server remounts HandleIssue and HandleBulkIssue at another route (for ajax), but it's the same code.

What if we only check this if User is present (not authorized_app)

Comment (sethvargo) [reply pkg/controller/issueapi/issue_batch.go]: I guess that's fine? I just really dislike when a code path behaves dramatically different based on some other data that's a few layers of indirection away.

File: pkg/controller/issueapi/issue_batch.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func (c *Controller) HandleBatchIssue() http.Handler {
 			return
 		}
 
-		if !membership.Can(rbac.CodeBulkIssue) {
+		if membership.User != nil && !membership.Can(rbac.CodeBulkIssue) {
```
Comment (sethvargo) [inline pkg/controller/issueapi/issue_batch.go:75]: I think you need to check if _membership_ is nil. Membership.User will never be nil.

Comment (whaught) [reply pkg/controller/issueapi/issue_batch.go]: Yep sorry. Users have RBAC, not AuthApps.

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**